authors/apps/articles/views/like_article_view.py

```diff
@@ -0,0 +1,115 @@
+from django.http import Http404
+from django.core import exceptions
+
+from rest_framework import status
+from rest_framework.generics import RetrieveUpdateAPIView
+
+from rest_framework.permissions import (
+    IsAuthenticatedOrReadOnly, IsAuthenticated
+)
+
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from ..models import Article, LikesDislikesModel
+from ..renderers import ArticleJSONRenderer
+from ..serializers import (
+    ArticleSerializer, LikeDislikeSerializer
+    )
+from ..exceptions import ArticleDoesNotExist
+from django.db.models import Avg
+from django.shortcuts import render, get_object_or_404
+
+
+class LikeView(APIView):
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ArticleJSONRenderer,)
+    serializer_class = LikeDislikeSerializer
+
+    def post(self, request, slug):
+        user = request.user
+        article_to_be_liked = get_object_or_404(Article, slug=slug)
+        check_if_article_is_liked = LikesDislikesModel.objects.filter(
+            article_id = article_to_be_liked.id,
+            user_liking_id = user.id
```

Comment:
This should be `user_linking_id = user.profile.id` since the `user_liking` field in the **LikesDislikes** model relates to the Profile model and not the User model. The same change is required on lines 39, 81 and 86.

Reply:
Let me get this fixed

Reply:
I have fixed this